### ISSUE TITLE
[BUG] Forecasting notebook in examples is buggy

### ISSUE THREAD

**Issue by Lovkush-A** [alan-turing-institute/sktime#790]
* Needs to be updated with latest changes from #757, e.g. changing of `scitype`s
* Adjustment of `window_length`s in gridsearch, as they are currently too small. I think it is as simple as replacing `initial_window` with `window_length` in SlidingWindowSplitter.

**To Reproduce**
Try running the forecasting notebook for yourself

**Comment by Lovkush-A** [comment]
I am happy to work on this, but I will do it after #757 is completed.

**Comment by mloning** [comment]
Thanks @Lovkush-A for raising this issue! 

Seems like the [CI test](https://github.com/alan-turing-institute/sktime/blob/329a937ff60adbf6c48971dd1f17f3fd3d26c197/.github/workflows/build-and-test.yml#L27) doesn't fail if it detects an error: https://github.com/alan-turing-institute/sktime/runs/2286754227

I believe we need to add `set -e` at the top of the `build_tools/run_examples.sh` file. As this affects all PRs, we shouldn't wait until #757 is merged. 

I'd appreciate a separate PR with the fix in `build_tools/run_examples.sh` and the failing notebooks.
